### PR TITLE
arm: Remove CMSIS from the tree and use a module

### LINKED ISSUE #23373: ARM: Move CMSIS out of main tree
## Introduction

This is a proposal to move the CMSIS out of the main Zephyr repository.

### Problem description

CMSIS is one of the two "hal" layers that are still part of the main Zephyr tree, residing inside the /ext folder. 

### Proposed change

We would like to remove the ARM CMSIS "hal" layer from the /ext directory of the Zephyr main tree and instead, place it in a "hal" repository in the Zephyrproject organization, which, ideally, will mirror the ARM CMSIS repository.

## Detailed RFC

The repository content could be fetched using the west tool, as we do for all other HAL layers.

### Dependencies

Not applicable.

## Alternatives
Keep having the CMSIS as part of the main zephyr repository.

### PR DESCRIPTION
**Pull request by carlescufi**
Use an external CMSIS repo instead of keeping a copy in ext/hal.

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

See: https://github.com/zephyrproject-rtos/cmsis/pull/1
Closes #23373

### CODE REVIEW

**Comment by ioannisg** [comment]
You must remove the CODEOWNERS entries

**Comment by carlescufi** [comment]
> You must remove the CODEOWNERS entries

done

**Review by tyrantlizard77 — APPROVED** [review]

**Review by nashif — APPROVED** [review]

File: ext/hal/README [hunk 1]
```diff
@@ -7,53 +7,3 @@ atmel
 The Atmel Software Framework (ASF) provides device header files and peripheral
```
Comment (galak) [inline ext/hal/README]: Why keep atmel, seems like you can remove this file.  since it doesn't mention altera.

Comment (carlescufi) [reply ext/hal/README]: ok, will remove it

Comment (carlescufi) [reply ext/hal/README]: done

**Review by galak — APPROVED** [review]

**Review by ioannisg — APPROVED** [review]

**Pull request merged**